### ISSUE TITLE
http proxy support

### ISSUE THREAD

**Issue by Serph91P** [anxdpanic/plugin.video.twitch#696]
### What Problem Does This Solve? 

Love the feature from Streamlink to add a http proxy. For me this would fix the issue because Telekom a German ISP does not like twitch an many other. So if I use a VPN or a proxy everything works fine again. 

------

### Suggest a Solution (optional)

Since I don't want to use a VPN to route everything from Kodi I would love if this plugin would add support for this. I mean you could go even further and add ttv-lol support to block adds but I thibk this would probably go to far.

------

**Comment by Liqianyu** [comment]
The Twitch add-on is the same as the YouTube add-on, and the video streaming seems to follow the Kodi proxy settings.I hope the devs can refine the proxy support for Twitch, which would obviously solve the geo-blocking issue with Twitch.
https://help.twitch.tv/s/article/geoblocking

It also helps combat Internet censorship against Twitch, a world where more than a billion Internet users don't have direct access to Twitch.
https://en.wikipedia.org/wiki/Twitch_(service)#Internet_censorship

**Comment by Liqianyu** [comment]
https://github.com/anxdpanic/script.module.python.twitch/blob/dbad1e5c45780e0b0b0a03f153f5c3bbdf110bea/resources/lib/twitch/scraper.py#L103
```python
            # Add hard-coded proxy here
            proxies = {
                'http': 'http://ip:port',
                'https': 'http://ip:port',
            }
            response = requests.request(method=method, url=url, headers=headers,
                                        json=json_body, data=data, verify=SSL_VERIFICATION, proxies=proxies)
``` 

This modification allows Twitch add-ons to use hard-coded proxies before developers add proxy functionality.

**Comment by apo86** [comment]
> This modification allows Twitch add-ons to use hard-coded proxies before developers add proxy functionality.

There hasn't been any active development of this addon in quite some time.
If you have the knowledge to develop this functionality yourself, give it a go. I can't speak for the repo maintainer, but so far they've always been willing and able to merge PRs here and submit new versions to the official Kodi repo.

**Comment by Liqianyu** [comment]
@apo86 
Yes, I have tried to refine the proxy support. But unfortunately I don't have the experience in developing Kodi add-ons and enough time to complete it.
I think it might be much easier for a skilled Kodi add-on developer.
I did notice that you have submitted multiple PRs for the project, so it would be great if you would be willing to implement proxy support in your spare time.

**Comment by apo86** [comment]
So far I've only done some bug fixes and never dealt with anything front end related. I have no idea how to implement a new configurable setting. Maybe if I'm super bored at some point, I'll try to figure it out, but don't count on it.

**Comment by Liqianyu** [comment]
> So far I've only done some bug fixes and never dealt with anything front end related. I have no idea how to implement a new configurable setting. Maybe if I'm super bored at some point, I'll try to figure it out, but don't count on it.

Okay, try to look into it when I have time. Maybe we can collaborate on it.
